fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1757,9 +1788,12 @@ private void updateIndexState(@Nonnull String indexName, byte[] indexKey, @Nonnu
             } else {
                 tr.set(indexKey, Tuple.from(indexState.code()).pack());
             }
-            recordStoreState.setState(indexName, indexState);
+            recordStoreState.updateAndGet(state -> {
+                state.setState(indexName, indexState);
+                return state;
+            });
```

Comment:
I think this is still just `recordStoreState(.get()).setState(indexName, indexState)`.

Reply:
No, for the same index, the specified approach and the written approach are equivalent. (It’s already backed with a concurrent hash map, so one side will win the race.)

As mentioned in another comment, in theory, two threads could try and set the record store state at once. As written, the only time this could happen would be if someone called this at the same time as the thing setting new indexes to write-only in `checkVersion`, but I could see keeping this as is provided there’s a comment somewhere explaining why. I think just the one comment in, say, `beginRecordStoreStateRead()` would suffice.

Reply:
Sure. Added a comment.

Reply:
Maybe a comment here saying that there is information in `beginRecordStoreStateRead()` as to why this is an `updateAndGet`